extensions/defaults/schema/groupversionresources/groupversionresources.go

```diff
@@ -0,0 +1,276 @@
+		Group:    "management.cattle.io",
+		Version:  "v3",
+		Resource: "clusterroletemplatebindings",
+	}
+
+	return clusterRoleTemplateBinding
+}
+
+func RoleTemplate() schema.GroupVersionResource {
+	roleTemplate := schema.GroupVersionResource{
+		Group:    "management.cattle.io",
+		Version:  "v3",
+		Resource: "roletemplates",
+	}
+
+	return roleTemplate
+}
+
+func ProjectRoleTemplateBinding() schema.GroupVersionResource {
+	projectRoleTemplateBinding := schema.GroupVersionResource{
+		Group:    "management.cattle.io",
+		Version:  "v3",
+		Resource: "projectroletemplatebindings",
+	}
+
+	return projectRoleTemplateBinding
+}
+
+func ResourceQuota() schema.GroupVersionResource {
+	resourceQuota := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "resourcequotas",
+	}
+
+	return resourceQuota
+}
+
+func Secret() schema.GroupVersionResource {
+	secret := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "secrets",
+	}
+
+	return secret
+}
+
+func Service() schema.GroupVersionResource {
+	service := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "services",
+	}
+
+	return service
+}
+
+func StorageClass() schema.GroupVersionResource {
+	storageClass := schema.GroupVersionResource{
+		Group:    "storage.k8s.io",
+		Version:  "v1",
+		Resource: "storageclasses",
+	}
+
+	return storageClass
+}
+
+func Token() schema.GroupVersionResource {
+	token := schema.GroupVersionResource{
+		Group:    "management.cattle.io",
+		Version:  "v3",
+		Resource: "tokens",
+	}
+
+	return token
+}
+
+func PersistentVolumeClaim() schema.GroupVersionResource {
+	persistentVolumeClaim := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "persistentvolumeclaims",
+	}
+
+	return persistentVolumeClaim
+}
+
+func PersistentVolume() schema.GroupVersionResource {
+	persistentVolume := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "persistentvolumes",
+	}
+
+	return persistentVolume
+}
+
+func Namespace() schema.GroupVersionResource {
+	namespace := schema.GroupVersionResource{
+		Group:    "",
+		Version:  "v1",
+		Resource: "namespaces",
+	}
+
+	return namespace
+}
+
+func Daemonset() schema.GroupVersionResource {
+	daemonset := schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "daemonsets",
+	}
+
+	return daemonset
+}
+
+func Deployment() schema.GroupVersionResource {
+	deployment := schema.GroupVersionResource{
+		Group:    "apps",
+		Version:  "v1",
+		Resource: "deployments",
+	}
+
+	return deployment
+}
+
+func Job() schema.GroupVersionResource {
+	job := schema.GroupVersionResource{
+		Group:    "batch",
+		Version:  "v1",
+		Resource: "jobs",
+	}
+
+	return job
+}
+
+func CronJob() schema.GroupVersionResource {
+	cronJob := schema.GroupVersionResource{
+		Group:    "batch",
+		Version:  "v1beta1",
+		Resource: "cronjobs",
+	}
+
+	return cronJob
+}
```

Comment:
Do these have to be separated? I think this file would get longer whenever a new GVR is used in the framework.

What do you think about making this a template? Something like this:


```suggestion
func NewSchemaGVR(g, v, r string) schema.GroupVersionResource {
	return schema.GroupVersionResource{
		Group:    g,
		Version:  v,
		Resource: r,
	}
}
```

Reply:
These would be constructor functions. They all seem like **functions** that return **fixed** values. So one way or another they are functions right now, right? So what I am trying to say is, is it expected to add a new function for every new GVR instead of having a constructor function?

Reply:
Yes, they are returning fixed values, they are the equivalent of a "const" @caliskanugur. The 2 concerns I have for axing them and putting a nice generic constructor  are:
1. They would no longer be equivalent in function to a "const". This is the case since it would be the equivalent of declaring each GVR in place wherever it is called. i.e NewSchemaGVR("x", "y", "z") is dynamic, not const and therefore there is no way to update/reuse a given GVR across the framework.
2. If we go with the above approach then I would vote on moving the constructor into the Schema directory since it would no longer be a "const".